ufo2fontir/src/source.rs

```diff
@@ -290,7 +299,73 @@ impl DesignSpaceIrSource {
 struct StaticMetadataWork {
     designspace_file: PathBuf,
     designspace: Arc<DesignSpaceDocument>,
-    glyph_order: Arc<Vec<String>>,
+    glyph_names: Arc<HashSet<String>>,
+}
+
+fn default_master(designspace: &DesignSpaceDocument) -> Option<&designspace::Source> {
+    // The master at the default location on all axes
+    // TODO: fix me; ref https://github.com/googlefonts/fontmake-rs/issues/22
+    warn!("Using an incorrect algorithm to determine the default master");
+    let default_location: DesignSpaceLocation = designspace
+        .axes
+        .iter()
+        .map(|a| (a.name.clone(), OrderedFloat::from(a.default)))
+        .collect();
+    designspace
+        .sources
+        .iter()
+        .find(|source| to_ir_location(&source.location) == default_location)
+}
+
+fn load_lib_plist(ufo_dir: &Path) -> Result<plist::Dictionary, WorkError> {
+    let lib_plist_file = ufo_dir.join("lib.plist");
+    if !lib_plist_file.is_file() {
+        return Err(WorkError::FileExpected(lib_plist_file));
+    }
+    plist::Value::from_file(&lib_plist_file)
+        .map_err(|e| WorkError::ParseError(lib_plist_file.clone(), format!("{}", e)))?
+        .into_dictionary()
+        .ok_or_else(|| WorkError::ParseError(lib_plist_file, "Not a dictionary".to_string()))
+}
+
+// Per https://github.com/googlefonts/fontmake-rs/pull/43/files#r1044596662
+fn glyph_order(
+    designspace: &DesignSpaceDocument,
+    designspace_dir: &Path,
```

Comment:
I actually think it shouldn't because that would inextricably tie it to the filesystem. It feels right to me to store both independently in places that do care about a filesystem source.